server/etcdserver/api/membership/cluster.go

```diff
@@ -530,8 +531,13 @@ func (c *RaftCluster) PromoteMember(id types.ID, shouldApplyV3 ShouldApplyV3) {
 	if c.v2store != nil {
 		mustUpdateMemberInStore(c.lg, c.v2store, c.members[id])
 	}
-	if c.be != nil && shouldApplyV3 {
-		unsafeSaveMemberToBackend(c.lg, c.be, c.members[id])
+	if c.be != nil {
+		if shouldApplyV3 {
+			unsafeSaveMemberToBackend(c.lg, c.be, c.members[id])
+		} else {
+			c.lg.Info("Forcibly apply member promotion", zap.String("member", fmt.Sprintf("%+v", *c.members[id])))
+			unsafeHackySaveMemberToBackend(c.lg, c.be, c.members[id])
```

Comment:
We can't do that. When `shouldApplyV3` is `false`, it means v2store is lag behind of v3store, so we should NOT compare v2store and v3store.

Note learner can only be promoted once, and the operation can't be reverted, so forcibly promotion should be safe.

Reply:
Just noticed that you removed your comment, to be clearer, we don't need to do the same check of the 2 conditions in `SyncLearnerPromotionIfNeeded`